pkg/controller/disruption/disruption.go

```diff
@@ -386,11 +386,7 @@ func (r *ReconcileMachineDisruption) getMachinesForMachineDisruptionBudget(mdb *
 	}
 
 	machines := &mapiv1.MachineList{}
-	listOptions := &client.ListOptions{
-		Namespace:     mdb.Namespace,
-		LabelSelector: sel,
-	}
-	err = r.client.List(context.TODO(), listOptions, machines)
+	err = r.client.List(context.TODO(), machines, client.InNamespace(mdb.Namespace), client.MatchingLabels(mdb.Spec.Selector.MatchLabels))
```

Comment:
can we change it to something like this
```
	sel, err := metav1.LabelSelectorAsSelector(mdb.Spec.Selector)
	if err != nil {
		return nil, err
	}
	if sel.Empty() {
		return nil, nil
	}
	listOptions := &client.ListOptions{
		Namespace:     mdb.Namespace,
		LabelSelector: sel,
	}

	machines := &mapiv1.MachineList{}
	err = r.client.List(context.TODO(), machines, client.UseListOptions(listOptions))
```

and it more correct, because `client.MatchingLabels(mdb.Spec.Selector.MatchLabels)` does not really represent the selector, so we drop some possibilities. It also will fix `disruption` unittests

Reply:
Done